grid-client/deployer/deployment_deployer.go

```diff
@@ -92,13 +97,17 @@ func (d *DeploymentDeployer) Deploy(ctx context.Context, dl *workloads.Deploymen
 func (d *DeploymentDeployer) BatchDeploy(ctx context.Context, dls []*workloads.Deployment) error {
 	newDeployments := make(map[uint32][]gridtypes.Deployment)
 	newDeploymentsSolutionProvider := make(map[uint32][]*uint64)
-
+	networkUsedIPs := make(map[string]*[]byte)
 	for _, dl := range dls {
 		if err := d.Validate(ctx, dl); err != nil {
 			return err
 		}
-
-		generatedDls, err := d.GenerateVersionlessDeployments(ctx, dl)
+		if _, ok := networkUsedIPs[dl.NetworkName]; !ok {
+			network := d.tfPluginClient.State.Networks.GetNetwork(dl.NetworkName)
+			usedIPs := network.GetUsedNetworkHostIDs(dl.NodeID)
+			networkUsedIPs[dl.NetworkName] = &usedIPs
```

Comment:
why do u need the networkUserIPs map ? and not use the usedIPs directly inside the loop to pass it to the GenerateVersionlessDeployments function ?

Reply:
because BatchDeploy takes multiple deployments that might be in different networks so we need to track which used IPs in which network